frontend/catalyst/jax_primitives.py

```diff
@@ -1377,19 +1440,30 @@ def _for_loop_lowering(
         if apply_reverse_transform:
             # iv = start + normalized_iv * step
             body_args[0] = AddIOp(start_val, MulIOp(body_args[0], step_val))
+
         body_args[0] = IndexCastOp(loop_index_type, body_args[0]).result
         result_from_elements_op = ir.RankedTensorType.get((), loop_index_type)
         from_elements_op = FromElementsOp(result_from_elements_op, body_args[0])
         body_args[0] = from_elements_op.result
 
-        # recursively generate the mlir for the loop body
+        # Re-order arguments in accordance with jax dynamic API convensions
```

Comment:
One-line comments like this are actually incredible helpful to quickly decipher the different steps when reading a function. I would encourage putting these wherever possible, especially the parts with low-level JAX code (which is hard to read).